skins/laika/tests/unit/store/membership_fee_store.spec.ts

```diff
@@ -0,0 +1,315 @@
+import { MembershipFee } from '@/view_models/MembershipFee';
+import { Validity } from '@/view_models/Validity';
+import {
+	markEmptyFeeAsInvalid,
+} from '@/store/membership_fee/actionTypes';
+import {
+	MARK_EMPTY_FEE_INVALID,
+	SET_FEE,
+	SET_FEE_VALIDITY,
+	SET_INTERVAL,
+	SET_INTERVAL_VALIDITY,
+	SET_TYPE,
+	SET_TYPE_VALIDITY,
+} from '@/store/membership_fee/mutationTypes';
+import each from 'jest-each';
+import moxios from 'moxios';
+
+function newMinimalStore( overrides: Object ): MembershipFee {
+	return Object.assign(
+		{
+			validity: {
+				fee: Validity.INCOMPLETE,
+				type: Validity.INCOMPLETE,
+			},
+			values: {
+				fee: '',
+				interval: '',
+				type: 'BEZ',
+			},
+		},
+		overrides
+	);
+}
+
+describe( 'MembershipFee', () => {
+
+	const validityCases = [
+		[ Validity.VALID, true ],
+		[ Validity.INVALID, false ],
+		[ Validity.INCOMPLETE, true ],
+	];
+
+	describe( 'Getters/feeIsValid', () => {
+		it( 'does not return invalid fee on initalization', () => {
+			expect( getters.feeIsValid(
+				newMinimalStore( {} ),
+				null,
+				null,
+				null
+			) ).toBe( true );
+		} );
+
+		each( validityCases ).it( 'converts validity types to boolean state (test index %#)',
+			( feeValidity, isValid ) => {
+				const state = {
+					validity: {
+						fee: feeValidity,
+						type: Validity.INCOMPLETE,
+					},
+				};
+				expect( getters.feeIsValid(
+					newMinimalStore( state ),
+					null,
+					null,
+					null
+				) ).toBe( isValid );
+			},
+		);
+	} );
+
+	describe( 'Getters/typeIsValid', () => {
+		it( 'does not return invalid payment type on initalization', () => {
+			expect( getters.typeIsValid(
+				newMinimalStore( {} ),
+				null,
+				null,
+				null
+			) ).toBe( true );
+		} );
+
+		each( validityCases ).it(
+			'returns the expected validity for a given type (test index %#)',
+			( typeValidity, isValid ) => {
+				const state = {
+					validity: {
+						fee: Validity.INCOMPLETE,
+						type: typeValidity,
+					},
+				};
+				expect( getters.typeIsValid(
+					newMinimalStore( state ),
+					null,
+					null,
+					null
+				) ).toBe( isValid );
+			},
+		);
+	} );
+
+	describe( 'Actions/markEmptyFeeAsInvalid', () => {
+		it( 'commits to mutation [MARK_EMPTY_FEE_INVALID]', () => {
+			const commit = jest.fn();
+			const action = actions[ markEmptyFeeAsInvalid ] as any;
+			action( { commit } );
+			expect( commit ).toBeCalledWith(
+				MARK_EMPTY_FEE_INVALID
+			);
+		} );
+	} );
+
+	describe( 'Actions/markEmptyValuesAsInvalid', () => {
+		it( 'commits to mutation [MARK_EMPTY_FIELDS_INVALID]', () => {
+			const context = {
+				commit: jest.fn(),
+				getters: {
+					'membership_fee/paymentDataIsValid': true,
+				},
+			};
+			const action = actions.markEmptyValuesAsInvalid as any;
+			action( context );
+			expect( context.commit ).toBeCalledWith(
+				'MARK_EMPTY_FIELDS_INVALID'
+			);
+		} );
+	} );
+
+	describe( 'Actions/setInterval', () => {
```

Comment:
I think the two tests here can be combined into one like in 
https://github.com/wmde/FundraisingFrontend/blob/wmde19/skins/laika/tests/unit/store/bankdata_store.spec.ts#L156